enhancements/ingress/host-network-configurable-ports.md

```diff
@@ -0,0 +1,297 @@
+This is done by allowing configuration of the ports used by the router pods within the Ingress Controller API specification.
+
+## Motivation
+
+In the current implementation, Ingress Controllers in `HostNetwork` strategy use standard ports for HTTP (80), HTTPS (443), and stats (1936) on the underlying nodes. 
+
+As a result, each Ingress Controller needs to be deployed on its own set of cluster nodes. It is impossible to reuse the same nodes for two Ingress Controllers.
+
+For use cases such as multi-tenancy, multi-stage clusters, this results in increasing the number of cluster nodes, not to meet performance requirements, but for the sole purpose of having nodes with available http, https and stats ports for a new Ingress Controller.
+
+
+### Goals
+
+The goal of this enhancement is to allow cluster nodes to host router pods for more than one Ingress Controller with `HostNetwork` endpoint strategy. 
+
+This is achieved by allowing each Ingress Controller to run on a different set of ports, thus solving the issue of concurrency on the same ports of a cluster node.
+
+The Ingress Controller's port number become configurable in the API specification for protocols:
+- HTTP
+- HTTPS
+- Stats
+
+### Non-Goals
+
+- Allowing configuration of custom ports on the ingress controllers that utilise any endpoint publishing strategy other than `HostNetwork`.
+
+## Proposal
+
+Configuring http, https and stats port for the ingress controllers that use the `HostNetwork` endpoint publishing strategy.
+
+
+Update the IngressController API by adding port configuration fields in `HostNetwork` field:
+
+```go
+    // HostNetworkStrategy holds parameters for the HostNetwork endpoint publishing
+    // strategy.
+    type HostNetworkStrategy struct {
+        // ...
+        // HTTPPort is the port on the host which should be used to listen for
+        // HTTP requests.
+        // Define this field when default port 80 is known to be used by another process.
+        // Please avoid specifying a port within the nodeport range, which defaults to 
+        // 30000-32767, but could be configured differently.
+        // +kubebuilder:validation:Optional
+        // +kubebuilder:validation:Maximum=65535
+        // +kubebuilder:validation:Minimum=1
+        // +kubebuilder:default=80
+        HTTPPort int `json:"HTTPPort"`
+
+        // HTTPSPort is the port on the host which should be used to listen for
+        // HTTPS requests.
+        // Define this field when default port 443 is known to be used by another process. 
+        // Please avoid specifying a port within the nodeport range, which defaults to 
+        // 30000-32767, but could be configured differently.
+        // +kubebuilder:validation:Optional
+        // +kubebuilder:validation:Maximum=65535
+        // +kubebuilder:validation:Minimum=1
+        // +kubebuilder:default=443
+        HTTPSPort int `json:"HTTPSPort"`
```

Comment:
It's not the API definition, so it's less important, however to match with the example below (and with the most API examples actually): JSON tag names are better to be lower camel case.